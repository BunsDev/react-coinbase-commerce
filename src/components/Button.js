// @flow
import * as React from 'react';
import './button.css';

type Props = {
  styled?: boolean,
  children?: React.Node,
  className?: string
};

export default class Button extends React.PureComponent<Props> {
  render(){
    const {styled, children, ...props} = this.props;
    return (
      <button {...props} className={styled ? 'coinbase-commerce-button' : props.className}>{children || 'Buy With Crypto'}</button>
    );
  }
}
