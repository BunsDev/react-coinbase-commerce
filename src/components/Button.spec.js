import React from 'react';
import Button from './Button';

test('it renders and matches its snapshot',  () => {
  const wrapper = shallow(
    <Button/>
  );

  expect(wrapper).toMatchSnapshot();
});

test('it renders and matches its snapshot when the style prop is given', () => {
  const wrapper = shallow(
    <Button styled={true}>Test ?!</Button>
  );

  expect(wrapper).toMatchSnapshot();
});
