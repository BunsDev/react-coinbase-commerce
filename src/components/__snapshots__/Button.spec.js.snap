// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders and matches its snapshot 1`] = `
<button>
  Buy With Crypto
</button>
`;

exports[`it renders and matches its snapshot when the style prop is given 1`] = `
<button
  className="coinbase-commerce-button"
>
  Test ?!
</button>
`;
